pkg/generate/app/pipeline.go

```diff
@@ -140,24 +141,24 @@ func (g PipelineGroup) String() string {
 	return strings.Join(s, "+")
 }
 
-const MaxServiceNameLen = 24
+const maxServiceNameLength = 24
 
-var InvalidServiceChars = regexp.MustCompile("[^-a-z0-9]")
+var invalidServiceChars = regexp.MustCompile("[^-a-z0-9]")
 
-func makeValidServiceName(name string) string {
-	name = strings.ToLower(name)
-	name = InvalidServiceChars.ReplaceAllString(name, "")
-	if len(name) == 0 {
-		return fmt.Sprintf("svc-%d", rand.Intn(100000))
-	}
-	if len(name) > MaxServiceNameLen-5 {
-		name = name[:MaxServiceNameLen-5]
+func makeValidServiceName(name string) (string, string) {
+	if ok, _ := validation.ValidateServiceName(name, false); ok {
+		return name, ""
```

Comment:
I think that even if the name is valid, we'd still want to use GenerateName b/c you can have multiple ports per container/deployment. (I got rid of the break you had before)